Source/Core/DolphinQt2/MainWindow.cpp

```diff
@@ -480,3 +485,44 @@ void MainWindow::SetStateSlot(int slot)
   Settings::Instance().SetStateSlot(slot);
   m_state_slot = slot;
 }
+
+void MainWindow::dragEnterEvent(QDragEnterEvent* event)
+{
+  if (event->mimeData()->hasUrls() && event->mimeData()->urls().size() == 1)
+    event->acceptProposedAction();
+}
+
+void MainWindow::dropEvent(QDropEvent* event)
+{
+  for (const QUrl& url : event->mimeData()->urls())
```

Comment:
In that case, can this only handle the first item in the list, then? Right now it looks like, if the check in `dragEnterEvent` was dropped, this would support drag-and-dropping multiple URLs.